app/src/main/java/com/easyrouteapp/MainActivity.java

```diff
@@ -0,0 +1,152 @@
+    private Toolbar searchToolbar;
+    private RoutesFragment fragRoutes;
+    private List<EntityBase> routes = new ArrayList<>();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        EventBus.getDefault().register(MainActivity.this);
+
+        searchToolbar = (Toolbar) findViewById(R.id.tb_search);
+        searchToolbar.setLogo(R.mipmap.ic_launcher);
+        setSupportActionBar(searchToolbar);
+
+        fragRoutes = (RoutesFragment) getSupportFragmentManager().findFragmentById(R.id.routes_fragment_container);
+        if (fragRoutes == null) {
+            fragRoutes = new RoutesFragment();
+            FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
+            ft.replace(R.id.routes_fragment_container, fragRoutes, "routesFrag");
+            ft.commit();
+        }
+        seachRoutesByQuery(getIntent());
+    }
+
+    @Override
+    public void onDestroy(){
+        super.onDestroy();
+        EventBus.getDefault().unregister(MainActivity.this);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+        SearchManager searchManager = (SearchManager) getSystemService(Context.SEARCH_SERVICE);
+        SearchView searchView;
+        MenuItem item = menu.findItem(R.id.action_searchable_activity);
+        searchView = getSearchView(item);
+        searchView.setSearchableInfo( searchManager.getSearchableInfo( getComponentName() ) );
+        searchView.setQueryHint(getResources().getString(R.string.search_hint));
+        addFloatButtonActionListener((FloatingActionButton) findViewById(R.id.fab));
+        return true;
+    }
+
+    private void addFloatButtonActionListener(FloatingActionButton floatButton) {
+        floatButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(!StatusNetworkConnectionHelper.verifyConnection(MainActivity.this) ) {
+                    Toast.makeText(getApplicationContext(), getResources().getString(R.string.message_without_con), Toast.LENGTH_LONG).show();
+                    return;
+                }
+                new GeocodingTask(getApplicationContext()).execute(getResources().getString(R.string.name_default_city_map));
+            }
+        });
+    }
+
+    private SearchView getSearchView(MenuItem item) {
+        if( Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB ){
+            return (SearchView) item.getActionView();
+        }
+        return (SearchView) MenuItemCompat.getActionView( item );
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onEventStartDetail(StartDetailRouteEvent event) {
```

Comment:
I chose use EventBus because the main functionality of this lib is help us with code decoupling. On this specific case, the click action on the RecycleView item is in the Adapter class, who doesn't know the parent Activity. With the use of post method we can redirect the action to any Activity which wanna open a instance of DetailActivity, isn't just about MainActivity receive this event. That is cool because allows a great code reusability.